src/support_sphere/lib/logic/cubit/signup_cubit.dart

```diff
@@ -16,61 +16,52 @@ class SignupCubit extends Cubit<SignupState> implements ValidatableCubit {
   final UserRepository _userRepository;
 
   void firstNameChanged(String value) {
-    emit(
-      state.copyWith(
-        givenName: value,
-      ),
-    );
+    emit(state.copyWith(givenName: value));
   }
 
   void lastNameChanged(String value) {
-    emit(
-      state.copyWith(
-        familyName: value,
-      ),
-    );
+    emit(state.copyWith(familyName: value));
   }
 
   void emailChanged(String value) {
-    emit(
-      state.copyWith(
-        email: value,
-      ),
-    );
+    emit(state.copyWith(email: value));    
   }
 
   void passwordChanged(String value) {
-    emit(
-      state.copyWith(
-        password: value,
-      ),
-    );
+    emit(state.copyWith(password: value));
   }
 
   void signupCodeChanged(String value) {
-    emit(
-      state.copyWith(
-        signupCode: value,
-      ),
-    );
+    emit(state.copyWith(signupCode: value));
   }
 
   void confirmedPasswordChanged(String value) {
-    emit(
-      state.copyWith(
-        confirmedPassword: value,
-      ),
-    );
+    emit(state.copyWith(confirmedPassword: value));
   }
 
   void toggleShowPassword() => changeShowPassword(emit, state);
+
   void setValid() => emit(state.copyWith(isValid: true));
   void setInvalid() => emit(state.copyWith(isValid: false));
 
+  bool isSignupButtonEnabled() {
+    return state.isValid && state.isAllFieldsFilled;
+  }
```

Comment:
See comment above for this, should be similar to the login one: https://github.com/uw-ssec/post-disaster-comms/pull/171/files#r1817430159

Reply:
updated